tests/Unit/Metadata/Service/ClassMetadataSearcherTest.php

```diff
@@ -163,6 +163,45 @@ public function testFindAllUsingElasticSearch(): void
         $this->assertSame('class1', $rawResult[0]['class']);
     }
 
+    public function testFindAllUsingElasticSearchWithEmptyClassProperties(): void
+    {
+        $property = $this->createMock(core_kernel_classes_Property::class);
+        $property->method('getRelatedClass')
+            ->willReturn(null);
+
+        $class = $this->createMock(core_kernel_classes_Property::class);
+        $class->method('getLabel')
+            ->willReturn('Class label');
+
+        $this->ontology
+            ->method('getProperty')
+            ->willReturn($property);
+
+        $this->ontology
+            ->method('getClass')
+            ->willReturn($class);
+
+        $this->advancedSearchChecker
+            ->method('isEnabled')
+            ->willReturn(true);
+
+        $this->elasticSearch
```

Comment:
Is it possible to validate the received parameters on the `search` method with the `with` method on PHPUnit?

Reply:
We can check the actual format sent to Elasticsearch as well as the index used. Changed in 4fe8a48da5b175979e8a357efd2d978f23df05f3 and ab25288302c40f042936af49f99110652b5244eb.